python/lsst/ip/isr/isrTask.py

```diff
@@ -1590,6 +1596,34 @@ def run(self, ccdExposure, *, camera=None, bias=None, linearizer=None,
                                    amp.getName(), qaStats.getValue(afwMath.MEDIAN),
                                    qaStats.getValue(afwMath.STDEVCLIP))
 
+        # Calculate standard image quality statistics
+        if self.config.doStandardStatistics:
+            metadata = ccdExposure.getMetadata()
+            for amp in ccd:
+                ampExposure = ccdExposure.Factory(ccdExposure, amp.getBBox())
+                ampName = amp.getName()
+                metadata[f"LSST ISR MASK SAT {ampName}"] = isrFunctions.countMaskedPixels(
+                    ampExposure.getMaskedImage(),
+                    [self.config.saturatedMaskName]
+                )
+                metadata[f"LSST ISR MASK BAD {ampName}"] = isrFunctions.countMaskedPixels(
+                    ampExposure.getMaskedImage(),
+                    ["BAD"]
+                )
+                qaStats = afwMath.makeStatistics(ampExposure.getImage(),
+                                                 afwMath.MEAN | afwMath.MEDIAN | afwMath.STDEVCLIP)
+
+                metadata[f"LSST ISR FINAL MEAN {ampName}"] = qaStats.getValue(afwMath.MEAN)
+                metadata[f"LSST ISR FINAL MEDIAN {ampName}"] = qaStats.getValue(afwMath.MEDIAN)
+                metadata[f"LSST ISR FINAL STDEV {ampName}"] = qaStats.getValue(afwMath.STDEVCLIP)
+
+                if self.config.doOverscan:
+                    k1 = f"LSST ISR FINAL MEDIAN {ampName}"
+                    k2 = f"LSST ISR OVERSCAN SERIAL MEDIAN {ampName}"
+                    metadata[f"LSST ISR LEVEL {ampName}"] = metadata[k1] - metadata[k2]
+                else:
+                    metadata[f"LSST ISR LEVEL {ampName}"] = numpy.nan
```

Comment:
What happens to `metadata` at the end of the loop? Shouldn't it be linked to `self.metadata` before the next iteration? Or are these available the next time `ccdExposure.getMetadata()` is called?

Reply:
They should be immediately available.  `getMetadata` gives a pointer (for lack of a better term on my part) to the exposure metadata, and repeated calls yield identical pointers:

```
exp = afwImage.ExposureF.readFits(filename)
mm1 = exp.getMetadata()
mm2 = exp.getMetadata()
print(repr(mm1), repr(mm2))
## <lsst.daf.base.propertyContainer.propertyList.PropertyList object at 0x7f60dc04f030> <lsst.daf.base.propertyContainer.propertyList.PropertyList object at 0x7f60dc04f030>
```

I've previously put information into `self.metadata` (the task's metadata), but I think that's always been a bad decision, and this is the start of fixing that.